src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/RunnableProjectGenerator.cs

```diff
@@ -151,84 +159,130 @@ public IList<ITemplate> GetTemplatesAndLangpacksFromDir(IMountPoint source, out
                         {
                             string locale = locFile.Name.Substring(LocalizationFilePrefix.Length, locFile.Name.Length - LocalizationFilePrefix.Length - LocalizationFileExtension.Length);
 
-                            if (TryGetLangPackFromFile(locFile, out ILocalizationModel? locModel) && locModel != null)
+                            try
+                            {
+                                ILocalizationModel locModel = LocalizationModelDeserializer.Deserialize(locFile);
+                                if (templateModel.VerifyLocalizationModel(locModel, out IEnumerable<string> errorMessages))
+                                {
+                                    localizations.Add(new LocalizationLocator(
+                                        locale,
+                                        locFile.FullPath,
+                                        templateModel.Identity,
+                                        locModel.Author ?? string.Empty,
+                                        locModel.Name ?? string.Empty,
+                                        locModel.Description ?? string.Empty,
+                                        locModel.ParameterSymbols));
+                                }
+                                else
+                                {
+                                    HandleLocalizationValidationError(logger, file, locFile, errorMessages);
+                                }
+                            }
+                            catch (Exception ex)
                             {
-                                localizations.Add(new LocalizationLocator(
-                                    locale,
-                                    locFile.FullPath,
-                                    template.Identity,
-                                    locModel.Author ?? string.Empty,
-                                    locModel.Name ?? string.Empty,
-                                    locModel.Description ?? string.Empty,
-                                    locModel.ParameterSymbols));
+                                logger.LogWarning(LocalizableStrings.LocalizationModelDeserializer_Error_FailedToParse, locFile.GetFullPath());
+                                logger.LogDebug("Details: {0}", ex);
                             }
                         }
                     }
+
+                    //issue here: we need to pass locale as parameter
+                    //consider passing current locale file here if exists
+                    RunnableProjectTemplate runnableProjectTemplate = new RunnableProjectTemplate(this, templateModel, null, hostConfigFile);
```

Comment:
@bekir-ozturk I forgot to address this issue - could you pleas take a look?

Reply:
Created work item https://github.com/dotnet/templating/issues/3255